spec.emu

```diff
@@ -882,6 +882,297 @@ contributors: Google, Ecma International
+          1. If _isLeap_ is *false*, then
+            1. Set _isLeap_ to *true*.
+          1. Else,
+            1. Set _isLeap_ to *false*.
+            1. Set _number_ to _number_ + 1.
+        1. Return ~invalid~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-calendardatearithmeticaltoiso" type="implementation-defined abstract operation">
+      <h1>
+        CalendarDateArithmeticalToISO (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+          _month_: an integer,
+          _day_: an integer,
+        ): either a normal completion containing an ISO Date Record or a throw completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It returns an ISO Date Record that, when converted to a Calendar Date Record with _calendar_ (for example, with CalendarISOToDate), contains the given _year_, _month_, and _day_ values in its [[Year]], [[Month]], and [[Day]] fields. If this is not possible, a RangeError is thrown.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _year_, _month_, and _day_ do not form a valid date in _calendar_, throw a RangeError exception.
+        1. Return a calendar-dependent ISO Date Record as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-calendardaysinmonth" type="implementation-defined abstract operation">
+      <h1>
+        CalendarDaysInMonth (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+          _month_: an integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets _year_ and _month_ as arithmetical values in the given _calendar_ and returns the number of days in the corresponding month.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-calendarmonthsinyear" type="implementation-defined abstract operation">
+      <h1>
+        CalendarMonthsInYear (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _year_: an integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets _year_ as arithmetical value in the given _calendar_ and returns the number of months in the corresponding year.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-regulatenonisoyearmonth" type="implementation-defined abstract operation">
+      <h1>
+        RegulateNonISOYearMonth (
```

Comment:
thought: I find this naming a bit confusing. In the original spec, "Regulate" is about applying overflow options, but here it's really doing a type of addition operation.

Reply:
I replaced `RegulateNonISOYearMonth` with `ConstrainMonthCodeToArithmeticalMonth` in 6abbe6e3ca1ca220e33f3a62cb474fe631a75495